staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -603,6 +610,46 @@ func (td *typeDiscoverer) discoverStruct(thisNode *typeNode, fldPath *field.Path
 					}
 				}
 			}
+		case types.Struct:
+			for _, subfield := range childType.Members {
```

Comment:
Good point, added a comment:
```
 			// To support inner validations, here we build up childNode types have a +k8s:inner tag reference
 			// so that we can later generate calls to field specific validators from the child.inner information.
```